pkg/queue/breaker.go

```diff
@@ -130,8 +135,17 @@ type semaphore struct {
 }
 
 // acquire receives the token from the semaphore, potentially blocking.
-func (s *semaphore) acquire() {
-	<-s.queue
+func (s *semaphore) acquire(timeout time.Duration) bool {
+	var timeoutCh <-chan time.Time
+	if timeout != 0 {
+		timeoutCh = time.After(timeout)
+	}
+	select {
+	case <-s.queue:
```

Comment:
I am a bit unhappy that if they trigger at the same time, we might get a timeout rather than a success, but I guess that's Go for you :-)

Reply:
Yeah, I don't think theres a great fix for that but also timeout should be >> when we are expecting an event to come in.